File: 349-IntersectionOfTwoArrays/349-IntersectionOfTwoArrays.py
# Last updated: 09/07/2026, 10:08:29
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        sort1 = sorted(nums1)
        sort2 = sorted(nums2)
        i, j = 0, 0
        result = set()

        while i < len(sort1) and j < len(sort2):
            if sort1[i] == sort2[j]:
                result.add(sort1[i])
                i += 1
                j += 1
            elif sort1[i] < sort2[j]:
                i += 1
            else:
                j += 1

        return list(result)


                

File: 349-IntersectionOfTwoArrays/349-IntersectionOfTwoArrays.java
// Last updated: 09/07/2026, 10:08:23
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        Arrays.sort(nums1);
        HashSet<Integer> set = new HashSet<>();

        for(int n : nums2){
            if(bs(nums1,n)){
                set.add(n);
            }
        }
          int[] result = new int[set.size()];
        int i = 0;
        for(int j : set){
            result[i++] = j;

        }
        return result;

    }
        private boolean bs(int[] arr , int t){
            int left = 0;
            int right = arr.length-1;
            while(left<=right){
                int mid = (left+right)/2;
                if(arr[mid] == t){
                    return true;
                }
                else if(arr[mid]<t){
                    left = mid+1;
                }
                else{
                    right = mid-1;
                }
            }
            return false;
        }

      
    
}